guides/common/modules/ref_customizing-the-registration-templates.adoc

```diff
@@ -21,9 +21,12 @@ You can configure the following global parameters by navigating to *Configure* >
 
 ifndef::satellite[]
 * The `host_registration_insights` parameter is used in the `insights` snippet, the default value is `false`.
+The type must be set to `Boolean` to override the parameter value.
 endif::[]
 ifdef::satellite[]
 * The `host_registration_insights` parameter is used in the `insights` snippet, the default value is `true`.
+The type must be set to `Boolean` to override the parameter value.
+If the parameter is set to `false`, it prevents hosts from appearing in the `rh-cloud reports` and also restricts `insights-client` from uploading its reports.
```

Comment:
```suggestion
If the parameter is set to `false`, it prevents hosts from appearing in the Red Hat Cloud Console reports and also the `insights-client` from uploading reports to {Project}.
```
I'm not sure, about Insights client uploading reports to {Project}, verify with a SME.
Also, this sentence is quite important and I think it should be included for both non-Satellite and Satellite alike.

Reply:
I can confirm. The code looks for this parameter and will reject `insights-client` reports if the parameter is set to `false`.

Reply:
As for `Red Hat Cloud Console reports`: I think it's a bit vague. There are two types of reports: one generated by insights-client on the host itself, and the other generated by the rh_cloud plugin. Both are contributing to the Red Hat Inventory. While `insights-client` reports are used to generate recommendations, rh_cloud reports are used for inventory and RHEL usage reporting.

Reply:
The word "restricted" is a bit confusing for few reason. Because the parameter influence the system during its deployment. A user can change the insights-client whenever they want after that. Setting the parameter to "false" just do not enable insights and therefore nothing is sent. But it does not restrict per say. 

insights-client do no send data to Satellite in a way to be consumed by it when the parameter is set to true, it uses Satellite as a proxy. 

This parameter is not something that influence Satellite itself, it influences what kickstart scripts are given to the client system getting deployed via Satellite.

Reply:
@cmarinea not only during the deployment. I have specifically added code that would prevent Foreman from forwarding reports from `insights-client` to the cloud: https://github.com/theforeman/foreman_rh_cloud/blob/foreman_3_7/app/controllers/insights_cloud/api/machine_telemetries_controller.rb#L76

Of course you are correct about its influence during the deployment, but it's additional functionality that we have added specifically to the rh_cloud plugin.

Reply:
@ShimShtein So just to confirm the parameter intended to influence the kickstart template file "host_registration_insights" has signification elsewhere in the code?

If a user enable insights-client later on, the Satellite will not act as a proxy?

For the rh_cloud plugin, I thought it was another configuration completely.  I may be a little confused :)

Reply:
> So just to confirm the parameter intended to influence the kickstart template file "host_registration_insights" has signification elsewhere in the code?

The parameter intended to regulate a feature: insights-client reports. This means that it will influence everything insights-client related, be it provisioning or reporting. Imagine a user who is using RHV templates to provision new machines (no provisioning is done through the kickstart process) , if the user sets the parameter to false, it would be rational that insights-client will not try to report to the cloud, even if it's part of the template.

> If a user enable insights-client later on, the Satellite will not act as a proxy?

As far as I know insights-client has a way to discover it consumes content from Satellite, and if it discovers this fact, it will use Satellite as a proxy. I think there is a way to prevent this automatic discovery process and point it directly to the cloud, but it is definitely not the default configuration.

> For the rh_cloud plugin, I thought it was another configuration completely. I may be a little confused :)

I have tried to reduce the amount of knobs and levers in rh_cloud, and if I identified an already existing way of expressing user's idea, I used it in the rh_cloud plugin. There are a couple of configuration options specific to rh_cloud, but those are there only because rh_cloud introduced new features that required its own configuration.

Reply:
Pushed. I hope we're closer if not good to go. :crossed_fingers: